libvast/src/system/explorer.cpp

```diff
@@ -139,25 +139,36 @@ explorer(caf::stateful_actor<explorer_state>* self, node_actor node,
             return true;
         return false;
       };
-      auto leaves = caf::get<record_type>(layout).leaves();
-      auto it = std::find_if(leaves.begin(), leaves.end(), is_timestamp);
-      if (it == leaves.end()) {
+      auto timestamp_leaf = std::optional<record_type::leaf_view>{};
+      for (auto&& leaf : caf::get<record_type>(layout).leaves()) {
```

Comment:
Is it not possible to use stl algorithms with `generator`?

Reply:
Only those that allow for iterator+sentinel pairs, but some require iterators of equal types.